src/pages/index.tsx

```diff
@@ -104,8 +104,20 @@ export default HomePage;
 export { injectSessionGSSP as getServerSideProps } from '@/ssr-utils';
 
 const Carousel = () => {
+  const [initialPage, setInitialPage] = useState<number>(0);
+
+  useEffect(() => {
+    setInitialPage(parseInt(localStorage.getItem('carousel') ?? '0'));
```

Comment:
```suggestion
    setInitialPage(parseInt(localStorage.getItem('carousel') ?? '0', 10));
```
Just explicitly setting the radix

Reply:
fixed